zds/tutorialv2/mixins.py

```diff
@@ -235,15 +235,28 @@ class SingleContentDetailViewMixin(SingleContentViewMixin, DetailView):
     """
 
     def get(self, request, *args, **kwargs):
-        self.object = self.get_object()
+        try:
+            self.object = self.get_object()
+        except RedirectToPublicVersion:
+            if self.object.is_public and self.object.sha_public:
+                return HttpResponsePermanentRedirect(self.get_public_object().get_absolute_url_online)
```

Comment:
Un oubli je suppose ? :)
```suggestion
                return HttpResponsePermanentRedirect(self.get_public_object().get_absolute_url_online())
```

Reply:
Oui, je l’avais corrigé sur l’un des appels et l’ai oublié sur l’autre, dans lequel ne passait pas le test ^^'